### PR TITLE
[Feature][Connector-V2][Google Sheets] Add Google Sheets option rules

### PR DESCRIPTION
**Pull request by TaoZex**
## Purpose of this pull request

subtask of https://github.com/apache/incubator-seatunnel/issues/3337
Add Google Sheets option rules

The schema has been modified.If google sheets is not modified, it will build failure.

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Comment by TaoZex** [comment]
@EricJoy2048 @Hisoka-X @TyrantLucifer PTAL

File: seatunnel-connectors-v2/connector-google-sheets/src/main/java/org/apache/seatunnel/connectors/seatunnel/google/sheets/source/SheetsSourceFactory.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        return "GoogleSheets";
+    }
+
+    @Override
+    public OptionRule optionRule() {
+        return OptionRule.builder()
+                .required(SheetsConfig.SERVICE_ACCOUNT_KEY)
+                .required(SheetsConfig.SHEET_ID)
+                .required(SheetsConfig.SHEET_NAME)
+                .required(SheetsConfig.RANGE)
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-google-sheets/src/main/java/org/apache/seatunnel/connectors/seatunnel/google/sheets/source/SheetsSourceFactory.java:41]: Please add optional `SeaTunnelSchema.SCHEMA`

Comment (TaoZex) [reply seatunnel-connectors-v2/connector-google-sheets/src/main/java/org/apache/seatunnel/connectors/seatunnel/google/sheets/source/SheetsSourceFactory.java:41]: Thanks for your reviw, I have fixed it.

File: seatunnel-connectors-v2/connector-google-sheets/src/main/java/org/apache/seatunnel/connectors/seatunnel/google/sheets/source/SheetsSourceFactory.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                .optional(SeaTunnelSchema.SCHEMA)
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-google-sheets/src/main/java/org/apache/seatunnel/connectors/seatunnel/google/sheets/source/SheetsSourceFactory.java:42]: `schema` is a required option.

Comment (TaoZex) [reply seatunnel-connectors-v2/connector-google-sheets/src/main/java/org/apache/seatunnel/connectors/seatunnel/google/sheets/source/SheetsSourceFactory.java:42]: c2b97708a75fd5f2b88f10c6951d206

If no schema is specified, the SimpleTextSchema is used by default.

Comment (Hisoka-X) [reply seatunnel-connectors-v2/connector-google-sheets/src/main/java/org/apache/seatunnel/connectors/seatunnel/google/sheets/source/SheetsSourceFactory.java:42]: In `GoogleSheets` are not required, @TaoZex Maybe you should update your doc to change schema's required value to `no`.
image

Comment (TaoZex) [reply seatunnel-connectors-v2/connector-google-sheets/src/main/java/org/apache/seatunnel/connectors/seatunnel/google/sheets/source/SheetsSourceFactory.java:42]: I have changed it,thx.

**Review by Hisoka-X — APPROVED** [review]
LGTM

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**